### PR TITLE
*: Support trim function

### PR DESCRIPTION
**Pull request by shenli**
See: https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_trim

### CODE REVIEW

File: expression/trim.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+// See: https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_trim
+type FunctionTrim struct {
+	Str       Expression
+	RemStr    Expression
+	Direction int
+}
+
+// Clone implements the Expression Clone interface.
+func (f *FunctionTrim) Clone() Expression {
+	nf := &FunctionTrim{
```
Comment (ngaut) [inline expression/trim.go]: s/nf/ft/g

File: expression/trim.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	}
+	if len(remstr) > 0 {
+		x := trimLeft(str, remstr)
+		x = trimRight(x, remstr)
+		return x, nil
+	}
+	return strings.Trim(str, "\n\t "), nil
+}
+
+func trimLeft(str, remstr string) string {
```
Comment (qiuyesuifeng) [inline expression/trim.go]: Can use TrimLeft instead?
Like `strings.TrimLeft("abcabcxxxabc", "abc")`, the result is `xxxabc`, is it what u want?

Comment (shenli) [inline expression/trim.go]: Try this: fmt.Println(strings.TrimLeft("abcacbxxx", "abc")). You will get xxx, not acbxxx.

File: expression/trim.go (hunk 1 above)
Comment (shenli) [inline expression/trim.go]: n is new for short. So nf is new f.

File: expression/trim.go (hunk 2 above)
Comment (qiuyesuifeng) [inline expression/trim.go]: gotcha.

Comment (qiuyesuifeng) [inline expression/trim.go]: How about 
```
func trimLeft(str, remstr string) string {
	for {
		if strings.HasPrefix(str, remstr) {
			str = str[len(remstr):]
			continue
		}

		return str
	}
}
```

Comment (shenli) [inline expression/trim.go]: This is ok. But I can not see the difference with my implementation.
TrimPrefix returns s without the provided leading prefix string. If s doesn't start with prefix, s is returned unchanged.

Comment (qiuyesuifeng) [inline expression/trim.go]: Yes, u are right, TrimPrefix source code also use HasPrefix.

**Comment by qiuyesuifeng** [comment]
LGTM

File: expression/trim.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	// Do trim
+	if f.Direction == TrimLeading {
+		if len(remstr) > 0 {
+			return trimLeft(str, remstr), nil
+		}
+		return strings.TrimLeft(str, "\n\t "), nil
```
Comment (siddontang) [inline expression/trim.go]: `\r` is space too.

**Comment by shenli** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

**Pull request merged**